### ISSUE TITLE
 Add the ability to view an edit history

### ISSUE THREAD

**Issue by K4LCIFER** [thunder-app/thunder#538]
### Is your feature request related to a problem? Please describe.

Currently the user is only able to see that a post *was* edited, and when, but cannot see *what* was changed. This carries substantial potential for abuse -- a user could make a post, and gain a large amount of engagement, and then subsequently change its content to mislead future readers, or nullify all responses altogether. This drastically diminishes a users trust in the authenticity of comments, and thus devalues Lemmy's content on the whole.

### Describe the solution you'd like

I propose a "git-diff" style solution very similar to how [Element](https://element.io/) does it, as shown below
![image](https://github.com/thunder-app/thunder/assets/7829231/a3d817d9-2efa-4fdf-a310-59daf2482357)

One would tap on the pencil icon, and be able to see what was changed in the post, or comment.

### Describe alternatives you've considered

n/a

### Additional context

This most likely would need backend changes for proper functionality. I have submitted a feature request [here](https://github.com/LemmyNet/lemmy/issues/3803) to add an edit history to the Lemmy backend.

**Comment by hjiangsu** [comment]
Setting this to P3 because lemmy does not currently support this at the moment

**Comment by Glazed** [comment]
And I hope Lemmy _never_ supports this. If I accidentally doxx myself in my post by giving a little too much personal info, I should be able to edit my post to remove that without having to resort to deleting the entire post. 

In an entire decade of using reddit, and now Lemmy, I have literally never seen someone abuse the edit feature to do what OP says.

**Comment by K4LCIFER** [comment]
> In an entire decade of using reddit, and now Lemmy, I have literally never seen someone abuse the edit feature to do what OP says.

I'm surprised. Especially since the reddit protests — because of the reddit protests, a ton of people started mass replacing their comments and posts with some comment about spez or their dislike for reddit or something else.

**Comment by Glazed** [comment]
You're right about federation and the internet in general, of course, and if I ever gave out too much information, it would be subtle so the danger would be minimal. But I'd still prefer to have the option of editing it permanently and hoping for the best instead of doing the opposite and making it trivially easy to see the pre-edit text.

Besides, deletes are federated exactly like edits. Deleting it is no better than editing it in that regard.

As to "why not delete?" The whole point of using a site like Lemmy is to communicate with people. If my post has a bunch of good replies, why would I want to nuke the whole thing just to delete a few words?

I haven't been back to reddit since the protests. I was one of the first to leave. I also consider that quite different than some troll making one comment and then pulling a switcheroo, which I have never seen. Not that I remember, at least. Trolls will be trolls, no matter what technique they use, and what do you even gain by seeing their pre-edit troll text anyway?

**Comment by micahmo** [comment]
I appreciate GitHub's approach to editing. By default, each revision is shown, and you can perform a diff to see exactly what changed. This is incredibly helpful, for example to see the history of a Pull Request (if the submitter frequently updates the original comment) or maybe because someone unintentionally removed some useful troubleshooting info, etc.

However, you can also choose to delete a particular revision, if it happens to contain something sensitive.

I think this is a reasonable approach. Of course, who knows if we'll ever get anything like that on Lemmy.

**Comment by K4LCIFER** [comment]
> However, you can also choose to delete a particular revision, if it happens to contain something sensitive.

I agree that this would be reasonable, just so long as it is shown that the edit was deleted. Something like "The user deleted this edit." for the corresponding entry would suffice. At least then it's obvious that something funky may have happened, and onlookers then have some form of a paper trail to follow.
